pkg/oauth/oauth.go

```diff
@@ -0,0 +1,452 @@
+		RefreshToken: tokens.RefreshToken,
+		CloudID:      cloudID,
+	}, nil
+}
+
+// LoadOAuthSecrets loads OAuth secrets from storage.
+func LoadOAuthSecrets(login string) (*OAuthSecrets, error) {
+	jiraDir, err := getJiraConfigDir()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Jira config directory: %w", err)
+	}
+	primaryStorage := utils.KeyRingStorage{User: login}
+	fallbackSecretStorage := utils.FileSystemStorage{BaseDir: jiraDir}
+	secrets, err := utils.LoadJSON[OAuthSecrets](primaryStorage, oauthSecretsFile)
+	if err != nil {
+		fmt.Printf("Warning: Primary storage failed to save, using fallback")
+		secrets, err = utils.LoadJSON[OAuthSecrets](fallbackSecretStorage, oauthSecretsFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load OAuth secrets: %w", err)
+		}
+	}
+
+	return &secrets, nil
+}
+
+// HasOAuthCredentials checks if OAuth credentials are present.
+func HasOAuthCredentials(login string) bool {
+	_, err := LoadOAuthSecrets(login)
+	return err == nil
+}
+
+// collectOAuthCredentials collects OAuth credentials from the user.
+func collectOAuthCredentials() (*OAuthConfig, error) {
+	answers := struct {
+		ClientID     string
+		ClientSecret string
+		RedirectURI  string
+	}{}
+
+	// Check for environment variables
+	envClientID := os.Getenv("JIRA_CLI_OAUTH_CLIENT_ID")
+	envClientSecret := os.Getenv("JIRA_CLI_OAUTH_CLIENT_SECRET")
+
+	q1 := &survey.Input{
+		Message: "Jira App Client ID:",
+		Help:    "This is the client ID of your Jira App that you created for OAuth authentication.",
+		Default: envClientID,
+	}
+
+	q2 := &survey.Password{
+		Message: "Jira App Client Secret:",
+		Help:    "This is the client secret of your Jira App that you created for OAuth authentication.",
+	}
+
+	q3 := &survey.Input{
+		Default: defaultRedirectURI,
+		Message: "Redirect URI:",
+		Help:    "The redirect URL for Jira App. Recommended to set as localhost.",
+	}
+
+	if envClientID == "" {
+		if err := survey.AskOne(q1, &answers.ClientID); err != nil {
+			return nil, err
+		}
+	} else {
+		answers.ClientID = envClientID
+	}
+	if envClientSecret == "" {
+		if err := survey.AskOne(q2, &answers.ClientSecret); err != nil {
+			return nil, err
+		}
+	} else {
+		answers.ClientSecret = envClientSecret
+	}
+	if err := survey.AskOne(q3, &answers.RedirectURI); err != nil {
+		return nil, err
+	}
+
+	return &OAuthConfig{
+		ClientID:     answers.ClientID,
+		ClientSecret: answers.ClientSecret,
+		RedirectURI:  answers.RedirectURI,
+		Scopes:       defaultScopes,
+	}, nil
+}
+
+// performOAuthFlow executes the OAuth authorization flow.
+func performOAuthFlow(config *OAuthConfig, httpTimeout time.Duration, openBrowser bool) (*oauth2.Token, error) {
+	fmt.Printf("Expected Scopes:\n")
+	for i, scope := range config.Scopes {
+		// Offline_access is a special scope to request a refresh token that wont be there to display in the JIRA UI, so we skip it.
+		if scope != "offline_access" {
+			fmt.Printf("  %2d. %s\n", i+1, scope)
+		}
+	}
```

Comment:
I tend to write something like this


```suggestion
	for i, scope := range config.Scopes {
		if scope == "offline_access" {
			// Offline_access is a special scope to request a refresh token that wont be there to display in the JIRA UI, so we skip it.
			continue
		}

		fmt.Printf("  %2d. %s\n", i+1, scope)
	}
```